### PR TITLE
split server and config states into different files

### LINKED ISSUE #45: refactor formula to be composable with external package states
Howdy!

For a long time, I've using Debian packaging on Debian Jessie to install ZooKeeper and this formula (a two year old version) to configure it. This was accomplished by manually excluding states that conflicted with the packaging. In addition, I was able to use `zookeeper:lookup:service_script` set to `False` to disable overwriting the init script that came with the packaging.

I tried updating my fork of the formula to make the`hosts_function` pillar available for business reasons. After trying my previous method, I was unable to get the (current version of the) formula to stop overriding the packaging (systemd unit file now shipped with Debian Stretch and config directory layout).

One of two changes could be made to facilitate this:

* (First intuition) Break `zookeeper-service` state into different states for `file.managed` and `service.running`. This will make the systemd unit file from this formula excludable without also excluding the service.
* Move server configuration states to a `zookeeper.config` states file. This can be composed with external states that setup packaging and the service. elasticsearch-formula has a state file like this.

I prefer the second choice and seek validation/support from someone else. I've already implemented a change, and it works well for the situation.

Thanks!

### PR DESCRIPTION
**Pull request by blbradley**
Fixes #45.

I've been using this for a while with no problems.

### CODE REVIEW

**Comment by blbradley** [comment]
Square and merge preferred due to messy commits.

**Comment by blbradley** [comment]
I'd like to do some testing some `zookeeper.confg` locally and document the state in the README before merging. But definitely looking for a review.

**Comment by sroegner** [comment]
@blbradley Can you help me understand what the motivation for the split is - to me most of what the server state _is_ is really what you moved to config. Is there any case in which it is useful to have the config exist independently/ of or without a server process?

**Comment by blbradley** [comment]
Yes! This change helps facilitate using native zookeeper Linux packaging (setup in private states repo) along with the formula's configuration features.

**Comment by sroegner** [comment]
Ah, right - thanks.

File: zookeeper/server/init.sls [hunk 1]
```diff
@@ -3,18 +3,7 @@
 
 include:
   - zookeeper
```
Comment (vutny) [inline zookeeper/server/init.sls]: Here I think `- zookeeper.config` is missing.

File: zookeeper/server/init.sls [hunk 2]
```diff
@@ -3,18 +3,7 @@
...
-    - user: root
-    - group: root
-
-zookeeper-data-dir:
-  file.directory:
-    - name: {{ zk.data_dir }}
-    - user: zookeeper
-    - group: zookeeper
-    - makedirs: True
+  - zookeeper.config
```
Comment (blbradley) [inline zookeeper/server/init.sls]: It is right here.

**Comment by blbradley** [comment]
@vutny It was at the end of a large diff block. So, I can see how you missed it.

Thanks! I still may have more changes, but I appreciate the approval of the general idea! @vutny @sroegner

**Comment by blbradley** [comment]
I have everything I want done. Looking for one more approval. Not sure if you approved @sroegner.

**Comment by blbradley** [comment]
Merged master because I'm following this PR on my Salt master.

**Comment by sroegner** [comment]
Verified on CentOS6, three-node cluster

**Comment by blbradley** [comment]
@sroegner Steffen! Thanks for your time. Have a lovely weekend!

**Pull request merged**